app/main.py

```diff
@@ -1,22 +1,90 @@
 import typing as tp
 import logging
 
-from flask import Flask, jsonify, Response
+from pydantic import ValidationError
 
 import settings
 
-logging.basicConfig(format='%(asctime)s - %(levelname)s - %(message)s', level=logging.INFO)
+from flask import Flask
+from flask_cors import CORS
+from flask_migrate import Migrate
+
+from utils.before_request import validate_body_for_update_create, authenticate
+from utils.data_process import CustomJSONEncoder
+
+from models import db
 
-app = Flask(__name__, instance_relative_config=True)
-app.config.from_mapping(
-    SECRET_KEY=settings.SECRET_KEY
+from utils.error_handlers import (
+    validation_error_handler,
+    empty_body_exception,
+    serializer_validation_error,
+    limit_offset_error_handler,
+    sort_error_handler,
 )
+from utils.exceptions import (
+    EmptyBodyException,
+    SerializerValidationError,
+    LimitOffsetError,
+    SortException,
+)
+
+logging.basicConfig(format='%(asctime)s - %(levelname)s - %(message)s', level=logging.INFO)
+
+
+def register_flask_application(config: tp.Any) -> Flask:
+    """
+    Register flask application
+    Returns:
+        current flask app
+    """
+
+    from auth.views import auth_urls
+    from account.views import account_urls
+    from video.views import video_urls
+
+    logging.info('Starting register application')
+    app: Flask = Flask(__name__, instance_relative_config=True)
+    app.config.from_mapping(**config)
+
+    # set custom json encoder
+    app.json_encoder = CustomJSONEncoder
+
+    # TODO it can be update to automatic
+    app.before_request(authenticate)
+    app.before_request(validate_body_for_update_create)
+
+    # register error handlers
+    app.register_error_handler(SortException, sort_error_handler)
+    app.register_error_handler(LimitOffsetError, limit_offset_error_handler)
+    app.register_error_handler(ValidationError, validation_error_handler)
+    app.register_error_handler(EmptyBodyException, empty_body_exception)
+    app.register_error_handler(SerializerValidationError, serializer_validation_error)
+
+    # TODO it can be update to automatic
+    app.register_blueprint(auth_urls)
+    app.register_blueprint(account_urls)
+    app.register_blueprint(video_urls)
+
+    CORS(app, resources={r"/*": {"origins": "*"}}, supports_credentials=True)
```

Comment:
* for origins is wrong. you should only allow requests from your frontend hosted on S3

Reply:
i know that i will change this latter sometimes i want to test from local